src/main/java/com/es/phoneshop/web/ProductListPageServlet.java

```diff
@@ -39,7 +40,7 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
 
         request.setAttribute("products", products);
 
-        var list = productDao.getRecentlyViewedProducts(request);
+        var list = new RecentlyViewedProductsService().getRecentlyViewedProducts(request);
```

Comment:
Create field RecentlyViewedProductsService in Servlet class

Reply:
Ok